taskgraph/Task.py

```diff
@@ -421,8 +426,10 @@ def _task_executor(self):
     def add_task(
             self, func=None, args=None, kwargs=None, task_name=None,
             target_path_list=None, ignore_path_list=None,
-            dependent_task_list=None, ignore_directories=True, priority=0,
-            hash_algorithm='sizetimestamp', copy_duplicate_artifact=False):
+            hash_target_files=True, dependent_task_list=None,
+            ignore_directories=True, priority=0,
+            hash_algorithm='sizetimestamp', copy_duplicate_artifact=False,
+            transient_run=False):

```

Comment:
YES! Originally I did what you suggested, but the problem is that we don't differentiate which files are hashed. When I say `exists` all I really care about is if the target path files exist. I can't think of a case where I want an `exists` on the target file but wouldn't want a regular hash on the input parameters. Hence the specific naming of `hash_target_files` defaulting to True. I hesitate to add an 'exists' hash algorithm to input files because I don't have a case where I've ever needed to do that. 

For what it's worth I've been using the `hash_target_files` syntax in some other projects and it feels initiative to me. 

So that's where that's at. I'm comfortable with this, but the next step would be to expand what `hash_algorithm` means and change it to some kind of per-argument parameter. I'd rather not do that until it's needed. What do you think?

Reply:
Ah, thanks for clarifying the intent here!  I forgot that the `hash_algorithm` can apply to the input files as well.  Let's keep this as it is!